backend-agent/main.py

```diff
@@ -216,6 +226,48 @@ def get_heatmap():
         return jsonify({'error': str(e)}), 500
 
 
+@app.route('/api/attacks', methods=['GET'])
+def get_attacks():
+    """
+    Endpoint to retrieve all attacks with their weights.
+    Returns a JSON object with attack names and their weights.
+    """
+    try:
+        attacks = db.session.query(Attack).all()
+        attack_list = [{'name': attack.name, 'weight': attack.weight}
+                       for attack in attacks]
+        return jsonify(attack_list), 200
+    except Exception as e:
+        return jsonify({'error': str(e)}), 500
+
+
+@app.route('/api/attacks', methods=['PUT'])
+def update_attack_weights():
+    """
+    Update weights for multiple attacks.
+    Expects a JSON object like: {"artPrompt": 2, "codeAttack": 1, ...}
+    """
+    verify_api_key()
+    try:
+        weights = request.get_json()
+        if not isinstance(weights, dict):
+            return jsonify({'error': 'Invalid payload format'}), 400
+
+        for name, weight in weights.items():
+            attack = db.session.query(Attack).filter_by(name=name).first()
```

Comment:
do we need to specify `name.lower()` or there is no risk the attack name is not lowercase and same as it appears in the db?

Reply:
attacks names are retrieved from db at this point (and they are lower cased when saved at first)

there is a (small) chance that the user (maliciously) add / modify the attack name manually when calling the endpoint
If it happens there is below a `if attack:` that checks that the attack exists in db, if it's not the case then it fails.

if we lower the attacks name it means we lower an added / modified entry I guess so I prefer that it fails tell me if it makes sense to you